detection_rules/rule_validators.py

```diff
@@ -235,63 +278,103 @@ def validate_stack_combos(self, data: QueryRuleData, meta: RuleMeta) -> Union[EQ
                 if exc:
                     raise exc
 
-    def validate_integration(self, data: QueryRuleData, meta: RuleMeta, package_integrations: List[dict]) -> Union[
-            EQL_ERROR_TYPES, None, ValueError]:
+    def validate_integration(self, data: QueryRuleData, meta: RuleMeta,
+                             package_integrations: List[dict]) -> Union[EQL_ERROR_TYPES, None, ValueError]:
         """Validate an EQL query while checking TOMLRule against integration schemas."""
         if meta.query_schema_validation is False or meta.maturity == "deprecated":
             # syntax only, which is done via self.ast
             return
 
         error_fields = {}
-        current_stack_version = ""
-        combined_schema = {}
-        for integration_schema_data in get_integration_schema_data(data, meta, package_integrations):
-            ecs_version = integration_schema_data['ecs_version']
-            integration = integration_schema_data['integration']
-            package = integration_schema_data['package']
-            package_version = integration_schema_data['package_version']
-            integration_schema = integration_schema_data['schema']
-            stack_version = integration_schema_data['stack_version']
-
-            if stack_version != current_stack_version:
-                # reset the combined schema for each stack version
-                current_stack_version = stack_version
-                combined_schema = {}
+        package_schemas = {}
+
+        # Initialize package_schemas with a nested structure
+        for integration_data in package_integrations:
+            package = integration_data["package"]
+            integration = integration_data["integration"]
+            if integration:
+                package_schemas.setdefault(package, {}).setdefault(integration, {})
+            else:
```

Comment:
```suggestion
            elif package and not integration:
```

To mirror the logic from line [133](https://github.com/elastic/detection-rules/pull/3372/files#r1445352817).